### ISSUE TITLE
Add more warnings about keeping computer on during name registration

### ISSUE THREAD

**Issue by jackzampolin** [blockstack/blockstack-browser#1052]
Users keep turning their computers off during the name registration phase. I think we can combat this in two ways:

- More warnings about this
- On browser restart we check to see if there is an active preorder and attempt to issue the second transaction if there is time before the `NAME_PREORDER` expires

This is a significant pain point for may users of our product and a significant negative experience right at the beginning of their relationship with it for those that run into the issue.

**Comment by jackzampolin** [comment]
After talking with @larrysalibra about this we came up with other potential solutions:

- Have an external service that completes these transactions
  * Pros: would work seamlessly, could be run locally?
  * Cons: creeping centralization
- Have the node.js server that serves the browser run a background task
  * Pros: Reuses existing infrastructure, would work on `browser.blockstack.org`
  * Cons: would potentially not work if computer went to sleep? Need to check on different OS
- OS specific solution

**Comment by yknl** [comment]
We should have an **optional** external service that can take care of the name registration without the need to keep the computer on. This way we can offer much better UX while allowing advanced users to complete the name registration via their own computer. Another advantage of the optional external service is that it might be possible to lower the transaction fees by waiting a bit longer.

**Comment by kantai** [comment]
> On browser restart we check to see if there is an active preorder and attempt to issue the second transaction if there is time before the NAME_PREORDER expires

That already is the case.

> Have an external service that completes these transactions.

This is doable, but would require some thinking to ensure that the external service never needs to have your keys (I'm pretty sure that it's doable though).

> Have the node.js server that serves the browser run a background task

This doesn't solve the problem of needing to have your computer on, but this is probably a good idea when we move the transaction generation code out of `blockstack-client`, because you still want a background process to do the second phase of the registration

*Alternative*

Allow users to chose to send registration as a zero-conf transaction following the preorder. Both transactions would hit the bitcoin network at the same time, and the UTXO from the preorder would ensure that they are ordered correctly. Only change needed here is that we have to make sure that the virtualchain would process this kind of transaction correctly.

**Comment by jackzampolin** [comment]
> Allow users to chose to send registration as a zero-conf transaction following the preorder. Both transactions would hit the bitcoin network at the same time, and the UTXO from the preorder would ensure that they are ordered correctly. Only change needed here is that we have to make sure that the virtualchain would process this kind of transaction correctly.

This I find to be the most attractive option

**Comment by shea256** [comment]
Yeah front-running is a theoretical issue.

Laptop closing is an actual issue.

Zero conf all the way. Let people check a box if they want to be protected from front-running.

**Comment by larrysalibra** [comment]
+1 to this.

We should also consider making the pre-order transaction optional in the next hard fork. If you're concerned about front-running, check the box and pay 2x the miner fees to broadcast a pre-order. If you're not, don't and get zero conf with 1/2 the price.

Related: other parts of the browser fail when the user doesn't have internet in unexpected ways: https://github.com/blockstack/blockstack-browser/issues/1077

**Comment by john-light** [comment]
I registered a name today and the browser didn't tell me anything about keeping my computer on/ keeping Blockstack Browser running. I closed the browser before the register tx was confirmed, but the registration still eventually went through correctly. What's going on here? Is the default behavior now to only send a register tx with no preorder tx? Or are both txs signed at once and now some central server is ensuring the second register tx is broadcast correctly? Or something else?
